src/main/java/org/folio/circulation/rules/Text2Drools.java

```diff
@@ -337,14 +337,23 @@ private void generateRule(PoliciesContext policies) {
     drools.append("end\n\n");
   }
 
-  private String policyMatchString(PolicyContext policy) {
+  private static String policyMatchString(PolicyContext policy) {
     String policyTypeName = "";
-    if (policy.POLICY_TYPE().toString().equals("l")) {
-      policyTypeName = "loan";
-    } else if (policy.POLICY_TYPE().toString().equals("r")) {
-      policyTypeName = "request";
-    } else if (policy.POLICY_TYPE().toString().equals("n")) {
-      policyTypeName = "notice";
+    switch (policy.POLICY_TYPE().toString()) {
+      case "l":
+        policyTypeName = "loan";
+        break;
+      case "r":
+        policyTypeName = "request";
+        break;
+      case "n":
+        policyTypeName = "notice";
+        break;
+      case "o":
+        policyTypeName = "overdue";
+        break;
+      default: throw new IllegalArgumentException("Unknown policy type: "
+        + policy.POLICY_TYPE().toString());
     }
     return String.format("    match.%sPolicyId = ", policyTypeName);
```

Comment:
This format() is somewhat difficult to read and understand. The name is broken into pieces: loanPolicyId is broken into loan and PolicyId. The pieces are not needed, they simply get concatenated.
Can you add some comment into the code why format() it is used, why the name is broken and why a more easy to read and understand and faster solution like

    case "l":
      return "    match.loanPolicyId = ";
    case "r":
      return "    match.requestPolicyId = ";
    …

is not used?

Reply:
@julianladisch  I actually don`t know why this approach was applied. Maybe it was used regarding Locale?

Reply:
@julianladisch Do you think that it`s better to remove this "String.format()" and use like this:
`case "l":
  return "    match.loanPolicyId = ";
case "r":
  return "    match.requestPolicyId = ";`  ?

Reply:
Yes.